### DISCUSSION TITLE
Everytime I am trying to import @react-native-firebase/auth emulator is throwing error that "The development server returned response error code :500"

### DISCUSSION THREAD

**Discussion by dipanshubahuguna** [invertase/react-native-firebase#4409] [Q&A] [answered]
Hi there,

I am using **rnfirebase** for **authentication** in my react-native project ,the emulator is throwing development server error everytime  while I am importing @react-native-firebase/auth in my App.js file .
I am using **react-native-cli** ,before that I was using expo-cli where I was using web configuration which didn't workout (As emulator was always throwing error that "module not installed in your android project by calling firebase.App()").So i switched to react-native-cli where i can use native dependencies for android and iOS respectively.
But here I am stuck now .
Please help me to solve this issue .
And one more thing if i wish to try this firebase thing using **yarn** will it be **good** for **future**??
![Screenshot 2020-10-17 140356](https://user-images.githubusercontent.com/54525722/96332443-a2051600-1081-11eb-8179-dee56f563995.jpg)

**Comment by mikehardy** [comment, marked as answer]
Hi there! Try this, it generates a basic project and demonstrates the integration steps required: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh

**Reply by dipanshubahuguna** [reply]
Nothing happened, still the emulator is throwing this development server error.
Whenever I remove the line (import auth from '@react-native-firebase/auth';) it works perfectly fine..
Any other suggestions ??
This error is really frustrating :(

**Reply by mikehardy** [reply]
"Nothing happened" not a lot to go on there. I promise the script works. Or it did last week anyway. Generates a full project and works, something is wrong with your project integration and you'll have to sift through the difference between what the demo script does and your project in order to find it

**Reply by dipanshubahuguna** [reply]
Hi there, 
Yesterday the same error was coming while I was using react-navigation ,and then I find out a command (react-native start --reset-cache)
and it worked out for me ,I was also thinking I might be doing some mistake in installing the dependencies.
But now this same error is due to firebase. 
The above demo script you provided is using yarn so should I go with yarn or  try it with npm ?

**Reply by mikehardy** [reply]
I used to use just npm, then for a while I used a mix. I only use yarn now and would never go back to npm. yarn wins on pure performance criteria, it is much faster. It is also seemingly just a bit more correct sometimes. In your case you may actually need `npx react-native-clean-project` more than an npm vs yarn change

**Reply by dipanshubahuguna** [reply]
Now there is no such error . The modules are installed correctly , but the emulator is showing error that "no firebase app '[default]' has been created - call firebase.initializeapp()" and I think the firebase is not connect with my project .
By the way Thanks man.
Really appreciate

**Reply by mikehardy** [reply]
With all respect, these are the failures on what should be the first steps of integration, you are having fundamental "how do I integrate a native module with react-native" problems. They are not really in scope for this project specifically. The demo script above shows exactly how to do it all and it works, which tells me react-native-firebase works, so as a maintainer here - with apologies - i have no support actions to take. It's working as intended and as documented and as demonstrated in an example

**Reply by dipanshubahuguna** [reply]
Now everything is working perfectly , the error was with my android studio environment setup that's why I was facing problems ..
After switching from expo-cli to RN bare it worked Finally!!     
Thanks for that script once again :)

**Reply by mikehardy** [reply]
Glad it helped! Good luck with your project - react-native is not easy to get all set up so it is building correctly everywhere but once you're back up to just working on JS screens with fast refresh it can move pretty quickly. Cheers
